docs/architecture/adr-017-single-binary-upgrade.md

```diff
@@ -0,0 +1,157 @@
+# ADR 017: Single binary sync
+
+## Status
+
+Proposed
+
+## Changelog
+
+- 2023/03/22: initial draft
+
+## Context
+
+It important to sync from genesis with one binary.
+It is needed for wallets, block explorers, indexers, analytics (e.g. Nansen).
+It makes it possible to sync headers from chain and download blobs from third party service.
+
+## Alternative Approaches
+
+Instead of having a single binary, we can have multiple binaries that are managed by another supervisory binary e.g. [Cosmovisor](https://docs.desmos.network/fullnode/cosmovisor/).
+Osmosis does it through having a significant number of binaries gathered in one docker container and switching them at particular heights.
+This approach has a lot of maintenance burden and is fragile to sync from genesis.
+
+## Decision
+
+TBD
+
+## Detailed Design
+
+- Cosmos blocks contain a `version.app` (version of state machine) that can be used to indicate which version of celestia-app should be used to process a block.
+  - Example: Osmosis is currently on [version.app: 15](https://www.mintscan.io/osmosis/blocks/8804111) and release [v15.0.0](https://github.com/osmosis-labs/osmosis/releases/tag/v15.0.0).
+
+  - Have to maintain old versions of binaries. Need to update dependencies for security issues.
+
+- Genesis has an original version.app. Does celestia-app need to explicitly define this?
+
+- Bitcoin / Ethereum leverage if statement upgrade logic. Downside to if statement logic in Cosmos ecosystem is if statements may not be relevant for new chains
+
+- UX issue to improve experience for node operators to upgrade
+
+- Maintenance issue for maintaining historical releases of binaries
+
+- What granularity should conditional logic be?
+  - App struct level
+  - Module level
+  - if/else on a line by line basis
+
+- Why app version and not block height?
+  - App version is bumped via upgrade module
+  - Depending on app version is universal across multiple networks.
+  - Block height will differ across testnets / chain-ids so depends on a `.toml` config file
+
+- Are we planning on upstream this?
+  - [CometBFT](https://github.com/cometbft/cometbft) is working on this for consensus.
+  - We’re working on Cosmos SDK portion
```

Comment:
are these just notes from the call? if so, perhaps we should clean this up and focus on the content of this ADR before merging